src/hssm/rl/likelihoods/builder.py

```diff
@@ -301,6 +301,46 @@ def _extract_array(source, idx):
     return collected
 
 
+def _validate_computed_parameters(
+    ssm_logp_func: AnnotatedFunction,
+    computed_params: list[str],
+) -> None:
+    """Validate that all computed parameters have corresponding compute functions.
+
+    Parameters
+    ----------
+    ssm_logp_func : AnnotatedFunction
+        The SSM log-likelihood function that should contain compute functions
+        for computed parameters.
+    computed_params : list[str]
+        List of parameter names identified as computed (not available in
+        data_cols, list_params, or extra_fields).
+
+    Raises
+    ------
+    ValueError
+        If computed parameters are identified but no compute functions are
+        provided, or if some computed parameters lack compute functions.
+    """
+    if not computed_params:
+        return
+
+    if not hasattr(ssm_logp_func, "computed") or not ssm_logp_func.computed:
+        raise ValueError(
+            f"Parameters {computed_params} are not available "
+            f"in data_cols, list_params, or extra_fields, but no compute "
+            f"functions are provided in ssm_logp_func.computed"
+        )
+
+    missing_compute_funcs = set(computed_params) - set(ssm_logp_func.computed.keys())
```

Comment:
This makes a lot of sense but not very computationally efficient. How about `missing_compute_funcs = [param for param in set(computed_params) if param not in ssm_logp_func.computed]`?

Mainly, I just don't like the creation of a set from the keys of a dict since it's already supposed to be a set